### ISSUE TITLE
HHVM crash when attempting to build a repo

### ISSUE THREAD

**Issue by fredemmott** [facebook/hhvm#8395]
### HHVM Version

3.29.1
2018-12-04 nightly
internal X release

### Operating System and Version

- FB Internal
- Ubuntu 16.04
- MacOS Mojave

### Standalone code, or other way to reproduce the problem

- go to a directory containing the HSL (either github, or flib/core/hack/lib)
- `/path/to/hhvm --hphp -o /tmp/hhout -t hhbc --input-dir $(pwd)`

### Actual result

FB:

```
running hphp...
parsing inputs...
analyzeProgram...
analyze all...
analyze all took 0'00" (4 us) wall time
processImportedLambdas...
processImportedLambdas took 0'00" (6 us) wall time
analyzeProgram took 0'00" (28 us) wall time
parsing inputs took 0'00" (288214 us) wall time
analyze final...
analyze final took 0'00" (3 us) wall time
creating binary HHBC files...
running HHBBC...

---------------------------------optimize_func----------------------------------
Optimizing:HH\Lib\Vec\range
hphp failed
running hphp took 0'47" (47659096 us) wall time
```

Core dump and stracktrace in /var/tmp/cores

OSS builds explicitly say 'segmentation fault'

### Expected result

A repo, or a meaningful error saying what we're doing wrong

**Comment by fredemmott** [comment]
t37650116

**Comment by azjezz** [comment]
+ Ubuntu 18 LTS

**Comment by fredemmott** [comment]
Fixed in nightlies, need to find commit and backport

**Comment by fredemmott** [comment]
Fixed for nightlies and 4.0; confirming that cherry-picking 7bb64899480158d92e202951b757eb1d1bca7fd8 is a suitable fix for 3.30
